lib/src/io/vm.dart

```diff
@@ -39,6 +41,14 @@ String readFile(String path) {
   }
 }
 
+Future<String> readStdin() async {
+  var completer = new Completer<String>();
+  var inputBuffer = new StringAccumulatorSink();
+  io.stdin.transform(new Utf8Decoder()).listen(inputBuffer.add,
+      onDone: () => completer.complete(inputBuffer.string));
+  return completer.future;
+}
```

Comment:
I think `Encoding.decodeStream()` does this.

And we should probably use `SystemEncoding` here as well.